Skip updating containers where no local image info can be retrieved

piksel opened the pull request:
Partially reverts containrrr/watchtower#571, since it causes `dockerClient.StartContainer()` to crash since it uses `imageInfo` to rebuild the config.

Fixes #584

piksel committed:
Revert "Image of running container no longer needed locally (#571)"

This reverts commit 6da66fb3129da691923bea59731733e3f06d30ac.

piksel committed:
Update client.go

tammert commented on pkg/container/client.go:
```diff
@@ -119,7 +119,12 @@ func (client dockerClient) GetContainer(containerID string) (Container, error) {
 	}
 
-	container := Container{containerInfo: &containerInfo}
+	imageInfo, _, err := client.api.ImageInspectWithRaw(bg, containerInfo.Image)
```
@simskij the crash found in #215 is actually caused by `client.api.ImageInspectWithRaw`, _not_ by `ContainerJSONBase.Image` -> `container.imageInfo.ID` needs `ImageInspectWithRaw`.

Alternatively, if we handle the error from `ImageInspectWithRaw` properly (a log message along the lines of "The source image for container x is not present locally, cant update, skipping...") we might have the most reasonable solution?

piksel replied:
I changed to a warning and then handle imageInfo being nil in `Update`, skipping the container in such an event.

piksel committed:
fix: skip updating when no image info can be retrieved

This will allow watchtower to continue even though the image info for a
container cannot be retrieved. If this happens one warning will be emitted
and the container will be skipped, unless NoRestart or OnlyMonitor is supplied

simonaronsson commented on internal/actions/update.go:
```diff
@@ -25,11 +26,13 @@ func Update(client container.Client, params types.UpdateParams) error {
+	for i, targetContainer := range containers {
+		stale, err := client.IsContainerStale(targetContainer)
+		if stale && !params.NoRestart && !params.MonitorOnly && !targetContainer.HasImageInfo() {
+			err = errors.New("container does not have any available image info and cannot be automatically restarted")
```
Mind changing this to an `Errorf` and include the container name like on line 35?

piksel replied:
Its handled just beneath and combined with the other situation where updating is skipped. I did shorten the message though, as it becomes quite verbose.

tammert reviewed (approved):
LGTM :+1:
